### PR TITLE
SLING-9658 Support binary archives in .far files

### PR DESCRIPTION
**Pull request by bosschaert**
New extension handler service API added.
InstallFeatureModelTask that uses these services to handle extensions.
Additionalservices are provided by the unpack extension to provide binary support.

### CODE REVIEW

File: bnd.bnd [hunk 1]
```diff
@@ -1,2 +1,2 @@
 -includeresource: @org.apache.sling.feature.extension.apiregions-[0-9.]*.jar!/org/apache/sling/feature/extension/apiregions/launcher/LauncherProperties.*
-Conditional-Package: org.apache.sling.feature.extension.apiregions.api
+Conditional-Package: org.apache.sling.feature.extension.apiregions.api,org.apache.felix.utils.manifest
```
Comment (cziegeler) [inline bnd.bnd]: Can we use the shade plugin to move these classes (the felix ones) to the sling namespace?

File: src/main/java/org/apache/sling/installer/factory/model/impl/FeatureModelInstallerPlugin.java [hunk 2]
```diff
@@ -102,6 +104,9 @@
 
     private final File storageDirectory;
 
+    @Reference
+    private volatile List<ExtensionHandler> extensionHandlers;
+
```
Comment (cziegeler) [inline src/main/java/org/apache/sling/installer/factory/model/impl/FeatureModelInstallerPlugin.java:111]: I assume "volatile" makes this dynamic?

**Review by cziegeler — COMMENTED** [review]
Lgtm in general, thanks

File: bnd.bnd (hunk 1 above)
Comment (bosschaert) [reply bnd.bnd]: Actually, while the code for a time used the felix manifest classes, it doesn't now any more, so I have reverted this changed line. Thanks for helping me spot that!

File: src/main/java/org/apache/sling/installer/factory/model/impl/FeatureModelInstallerPlugin.java (hunk 2 above)
Comment (bosschaert) [reply src/main/java/org/apache/sling/installer/factory/model/impl/FeatureModelInstallerPlugin.java:111]: Yes, as per Declarative Services spec. However I forgot that it's not greedy by default. So I added explicit selectors for this now.

**Comment by bosschaert** [comment]
Thanks for the review @cziegeler - I have made some updates to address your comments.

**Pull request merged**